canvas/services/event.py

```diff
@@ -52,6 +44,27 @@ def get_question_stats(question):
                 answers[answer] = 0
             answers[answer] += 1
 
+    correct_submissions_users = set()
+    partially_correct_submissions_users = set()
+    incorrect_submissions_users = set()
+
+    student_submissions = submissions.filter(uqj__user__role="Student").values(
+        "uqj__user_id", "is_correct", "is_partially_correct"
+    )
+
+    for submission in student_submissions:
+        if submission["is_correct"]:
+            correct_submissions_users.add(submission["uqj__user_id"])
+        elif submission["is_partially_correct"]:
+            partially_correct_submissions_users.add(submission["uqj__user_id"])
+        else:
+            incorrect_submissions_users.add(submission["uqj__user_id"])
+
+    partially_correct_submissions_users = partially_correct_submissions_users.difference(correct_submissions_users)
+    incorrect_submissions_users = incorrect_submissions_users.difference(
+        correct_submissions_users, partially_correct_submissions_users
+    )
+
```

Comment:
I still feel like this logic has flaws. Don't you include 2 entities for 1 user that made 2 submissions?

Reply:
I think you should use uqj's here. get all the uqjs for the course and then you can filter them based on solved, partially solved, etc.